### ISSUE TITLE
I think this code can not be convergence by training

### ISSUE THREAD

**Issue by mshmoon** [eriklindernoren/PyTorch-YOLOv3#112]
I have implement an exeperment.I use tiny-yolo and yolo to train about 1000 images on coco dataset,but after 30 epochs that the models of tiny-yolo and yolo can not be overfitting, so I think this code can not be convergence on the whole coco dataset.I guess the loss function have problem

**Comment by fengxinjie** [comment]
> I have implement an exeperment.I use tiny-yolo and yolo to train about 1000 images on coco dataset,but after 30 epochs that the models of tiny-yolo and yolo can not be overfitting, so I think this code can not be convergence on the whole coco dataset.I guess the loss function have problem

change:
pred_cls = torch.sigmoid(prediction[..., 5:]) # Cls pred.
to:
pred_cls = prediction[..., 5:] # Cls pred.

sigmoid is not need for nn.CrossEntropyLoss()

**Comment by mshmoon** [comment]
Thank you ,I will try it

**Comment by mshmoon** [comment]
> > I have implement an exeperment.I use tiny-yolo and yolo to train about 1000 images on coco dataset,but after 30 epochs that the models of tiny-yolo and yolo can not be overfitting, so I think this code can not be convergence on the whole coco dataset.I guess the loss function have problem
> 
> change:
> pred_cls = torch.sigmoid(prediction[..., 5:]) # Cls pred.
> to:
> pred_cls = prediction[..., 5:] # Cls pred.
> 
> sigmoid is not need for nn.CrossEntropyLoss()

I figure out this bug,but it is still can not be convergence.The test result is full of preson bounding box

**Comment by mshmoon** [comment]
> > I have implement an exeperment.I use tiny-yolo and yolo to train about 1000 images on coco dataset,but after 30 epochs that the models of tiny-yolo and yolo can not be overfitting, so I think this code can not be convergence on the whole coco dataset.I guess the loss function have problem
> 
> change:
> pred_cls = torch.sigmoid(prediction[..., 5:]) # Cls pred.
> to:
> pred_cls = prediction[..., 5:] # Cls pred.
> 
> sigmoid is not need for nn.CrossEntropyLoss()

Do you have ever train the model use this code

**Comment by Lechatelia** [comment]
i train this model on my dataset with about 16K images， and it  convergenced

i used the original code
_p[..., 4] = torch.sigmoid(p[..., 4])  # p_conf_

**Comment by mshmoon** [comment]
> i train this model on my dataset with about 16K images， and it convergenced
> 
> i used the original code
> _p[..., 4] = torch.sigmoid(p[..., 4]) # p_conf_

How is the test reults

**Comment by fourth-archive** [comment]
I found the repo below trains to convergence well when starting COCO training from scratch. It also implements automatic image augmentation, and approaches the darknet mAP after about 65 epochs:
https://github.com/ultralytics/yolov3

**Comment by ming71** [comment]
> I found the repo below trains to convergence well when starting COCO training from scratch. It also implements automatic image augmentation, and approaches the darknet mAP after about 65 epochs:
> https://github.com/ultralytics/yolov3

But why I always failed to train on my own datasets?
According to the describtion , COCO format label should be used to image , but I found it not okay , its code shows that normalized input labels are required instead of coco format!!!why?
(refer to my descrebtion here  :   https://github.com/ultralytics/yolov3/issues/95)

**Comment by eriklindernoren** [comment]
The latest version should converge. That said there are still some augmentation and other training tricks to be implemented. `ultralytics` version is a good option.
